### DISCUSSION TITLE
keep the model and the workflow in-sync.

### DISCUSSION THREAD

**Discussion by chandramohank** [microsoft/RulesEngine#88] [Q&A] [answered]
If I have multiple workflows on a single model (class), If one of the property name of a model changes , then I need to change all the workflows which are referring that property.

Is there any better way of handling it?. How do we keep the model and the workflow properties in-sync at one place?.

**Comment by abbasc52** [comment, marked as answer]
Currently it is not possible with RulesEngine. 

This problem is actually prevalent in any object model mapping. 
The simplest workaround would be to maintain 2 classes:- 
1. class1 which maps directly to the workflow
2. class2 which has frequent property name changes and then mapping it to class1 before passing to RulesEngine(RE)

This process could have been integrated in RE if RE offered globalParams at workflow level like we offer [localParams](https://github.com/microsoft/RulesEngine/wiki/Getting-Started#localparams) at rule level. But currently that feature is under review and has not been implemented yet.
